model/cluster_utility_test.go

```diff
@@ -0,0 +1,179 @@
+package model
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestSetUtilityVersion(t *testing.T) {
+	u := &utilityVersions{
+		Prometheus: "",
+		Nginx:      "",
+		Fluentbit:  "",
+	}
+
+	setUtilityVersion(u, PROMETHEUS, "1.9")
+	require.Equal(t, u.Prometheus, "1.9")
+
+	setUtilityVersion(u, NGINX, "0.9")
+	require.Equal(t, u.Prometheus, "1.9")
+	require.Equal(t, u.Nginx, "0.9")
+
+	setUtilityVersion(u, "an_error", "9")
+	require.Equal(t, u.Prometheus, "1.9")
+	require.Equal(t, u.Nginx, "0.9")
+}
+
+func TestGetUtilityVersion(t *testing.T) {
+	u := &utilityVersions{
+		Prometheus: "4",
+		Nginx:      "5",
+		Fluentbit:  "6",
+	}
+
+	require.Equal(t, getUtilityVersion(u, PROMETHEUS), "4")
+	require.Equal(t, getUtilityVersion(u, NGINX), "5")
+	require.Equal(t, getUtilityVersion(u, FLUENTBIT), "6")
+	require.Equal(t, getUtilityVersion(u, "anything else"), "")
+}
+
+func TestSetActualVersion(t *testing.T) {
+	c := &Cluster{
+		Provider:            "aws",
+		Provisioner:         "kops",
+		ProviderMetadata:    []byte(`{"provider": "test1"}`),
+		ProvisionerMetadata: []byte(`{"provisioner": "test1"}`),
+		AllowInstallations:  false,
+	}
+
+	require.Equal(t, c.UtilityMetadata, []byte(nil))
+	err := c.SetUtilityActualVersion(NGINX, "1.9.9")
+	require.NoError(t, err)
+
+	require.NotEqual(t, []byte(nil), c.UtilityMetadata)
+
+	version, err := c.ActualUtilityVersion(NGINX)
+	require.NoError(t, err)
+	require.Equal(t, "1.9.9", version)
```

Comment:
nit: I prefer to use `require` only for steps in the test that must pass to properly test the next step or to prevent a panic and `assert` when doing the check you're actually focused on. That way you have a chance to see multiple failures in one run of the test, instead of having to run, fix, run, fix, etc.

Reply:
Oh okay, sounds good!